bpf/lib/ipv4.h

```diff
@@ -111,7 +112,12 @@ ipv4_frag_register_datagram(struct __ctx_buff *ctx, int l4_off,
 	if (ret < 0)
 		return ret;
 
-	map_update_elem(&IPV4_FRAG_DATAGRAMS_MAP, frag_id, ports, BPF_ANY);
+	if (map_update_elem(&IPV4_FRAG_DATAGRAMS_MAP, frag_id, ports, BPF_ANY) < 0)
+		if (dir == CT_INGRESS || dir == CT_EGRESS)
```

Comment:
This check is not needed? You already have it in `ct_to_metrics_dir()`.

Reply:
I think we still need this check as I'm guarding against the case where `dir` is equal to [`CT_SERVICE`](https://github.com/cilium/cilium/blob/master/bpf/lib/common.h#L536) (otherwise we may and up emitting a metric with the direction set to 0, which would be invalid)

Reply:
Ok fair enough

Reply:
Why are we ignoring all `CT_SERVICE` packets? They do correspond to fragmented ingressing and egressing packets, right? If they are already counted elsewhere, then do we even need to update the `IPV4_FRAG_DATAGRAMS_MAP` map?